src/sentry/remote_config/docs/protocol.md

```diff
@@ -22,17 +22,66 @@ Retrieve a project's configuration.
 
 **Feature Object**
 
-| Field | Type   | Description                        |
-| ----- | ------ | ---------------------------------- |
-| key   | string | The name used to lookup a feature. |
-| value | any    | A JSON value.                      |
+| Field    | Type           | Description                                                                     |
+| -------- | -------------- | ------------------------------------------------------------------------------- |
+| key      | string         | The name used to lookup a feature.                                              |
+| value    | any            | A JSON value.                                                                   |
+| variants | array[Variant] | Features can have any number of possible variants which alter the return value. |
+
+Features may specify zero or more variants which may alter the evaluation result.
+
+**Variant Object**
+
+| Field   | Type        | Description                                                                               |
+| ------- | ----------- | ----------------------------------------------------------------------------------------- |
+| rollout | Rollout     | An object which contains instructions for rolling out a variant.                          |
+| rules   | array[Rule] | An array of rules which all must evaluate to true in order to return the variant's value. |
+| value   | any         | The value of the feature if the variant is successfully evaluated.                        |
+
+When evaluating a set of rules and varaints each rule is a logical `AND` while each variant is a logical `OR`. Failure to evaluate one variant does not prevent another variant from successfully evaluating. Failure to evaluate one rule does invalidate the remaining rules and the evaluation proceedure can eagerly terminate.
+
+**Rollout Object**
+
+| Field         | Type                | Description                                                 |
+| ------------- | ------------------- | ----------------------------------------------------------- |
+| percentage    | number              | The percentage of requests which should evaluate to true.   |
```

Comment:
idk, something like this:

```suggestion
| percentage    | number              | The percentage of requests which should evaluate to true. Valid values are `0.0` to `1.0`, increments of `0.001`  |
```

Reply:
I'm not sure I'm ready to commit to the range/step just yet.  I was thinking 0-100 as integer values.  Decimal precision seems too granular.  Happy to be convinced otherwise.  lmk.

Reply:
I think 1-100 is too narrow, should be a range of 1,000 buckets, or 10,000 buckets imo.

Decimal or integer i don't mind. There'll be a UI that fixes the range and steps for user benefit. SDK just needs to know if it's a float or integer so it can multiple/divide as needed.

Reply:
Right I agree we can fix this in UI.  From a product perspective I'm trying to understand the utility of rolling out to 50% versus 50.01%.  Not hard to do from a protocol perspective at all.  We can support it on a "just in case" basis.  But I was just curious of the necessity.

Reply:
bigger apps need more granularity

Reply:
Yeah I could see "lets start at 0.001% and move up gradually from there".  So maybe the correct approach is 0-100 with arbitrary decimal precision?

Reply:
Don't forget to refresh and see my previous comment :D

Reply:
I saw. i was typing at the same time and was just like..... ehh i'll post what i have

i think we're on the same page though. arbitrary precision works for me for now! we can revisit that if we have reason to in the future.